File: Android-My-Nanny-old-version/app/src/main/java/com/vald3nir/mynanny/views/splash/SplashActivity.java
package com.vald3nir.mynanny.views.splash;

import android.content.Intent;
import android.os.Bundle;

import com.vald3nir.mynanny.R;
import com.vald3nir.mynanny.views.custom.CustomActivity;
import com.vald3nir.mynanny.views.login.LoginActivity;

public class SplashActivity extends CustomActivity {

    private SplashDelegate delegate;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        delegate = new SplashDelegate();
    }

    @Override
    protected void onResume() {
        super.onResume();
        delegate.connectMQTT(this);
    }

    public void callLogin() {
        startActivity(new Intent(this, LoginActivity.class));
    }

}

File: app/src/main/java/com/vald3nir/my_nanny/presentation/splash/SplashActivity.kt
package com.vald3nir.my_nanny.presentation.splash

import android.os.Bundle
import com.vald3nir.my_nanny.common.core.BaseActivity
import com.vald3nir.my_nanny.databinding.ActivitySplashBinding
import org.koin.androidx.viewmodel.ext.android.viewModel

class SplashActivity : BaseActivity() {

    private lateinit var binding: ActivitySplashBinding
    private val viewModel: SplashViewModel by viewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySplashBinding.inflate(layoutInflater)
        setContentView(binding.root)
        viewModel.view = this
        initViews()
    }

    private fun initViews() {
        binding.loading.showLoadingAnimation()
        viewModel.checkUserLogger()
    }

}